File: lms_mobile_web/lib/features/admin/leave/domain/models/leave_type.dart
/// 휴가 종류
enum LeaveType {
  /// 연차
  annual('ANNUAL', '연차'),

  /// 병가
  sick('SICK', '병가'),

  /// 개인 사유
  personal('PERSONAL', '개인 사유'),

  /// 경조사
  family('FAMILY', '경조사'),

  /// 기타
  other('OTHER', '기타');

  final String value;
  final String displayName;

  const LeaveType(this.value, this.displayName);

  static LeaveType fromString(String value) {
    return LeaveType.values.firstWhere(
      (type) => type.value == value,
      orElse: () => LeaveType.other,
    );
  }

  @override
  String toString() => displayName;
}

File: domain/src/main/kotlin/com/lms/domain/model/leave/LeaveType.kt
package com.lms.domain.model.leave

/**
 * 휴가 유형
 */
enum class LeaveType(val description: String, val requiresApproval: Boolean) {
    ANNUAL("연차", true),
    SICK("병가", true),
    PERSONAL("개인 사유", true),
    MATERNITY("출산 휴가", true),
    PATERNITY("육아 휴가", true),
    BEREAVEMENT("경조사", true),
    UNPAID("무급 휴가", true)
}
